File: tests/data-conversion/array-object/newarr.ts
function TestArrayObject() {
  const tenElements = new Array<number>(10);
  const oneElement = new Array<number>(1.23);
  const threeElements = new Array(1, 2, 3);

  const arrayOfOneElement = Array.of(10);

  Array.isArray(tenElements);
  Array.isArray(123);
}

File: tests/data-conversion/array-object/newarr.cpp
#include "runtime/array.h"

namespace {

void TestArrayObject() {
  compilets::Array<double>* tenElements = compilets::MakeObject<compilets::Array<double>>(10);
  compilets::Array<double>* oneElement = compilets::MakeObject<compilets::Array<double>>(1.23);
  compilets::Array<double>* threeElements = compilets::MakeObject<compilets::Array<double>>(1, 2, 3);
}

}  // namespace